core/sql/cli/CliExtern.cpp

```diff
@@ -6316,7 +6316,8 @@ Lng32 SQL_EXEC_DeleteHbaseJNI()
       threadContext->incrNumOfCliCalls();
 
       HBaseClient_JNI::deleteInstance();
-      HiveClient_JNI::deleteInstance();
+      // The Hive client persists across connections
+      // HiveClient_JNI::deleteInstance();
```

Comment:
Good question, and I think any security issues are (or need to be) handled in the HiveMetaData object, which already persists between connections. The object that now also persists is HiveClient_JNI (and the corresponding Java object). Those don't keep a cache or state about Hive metadata, they just have methods to read and validate Hive metadata. So, in short, no, I don't believe there is a security concern with this change.